src/transformStateWithClones.js

```diff
@@ -7,7 +7,41 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let copyState = { ...state };
+  const versionsOfState = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        const { extraData } = action;
+
+        Object.assign(copyState, extraData);
+        break;
+
+      case 'removeProperties':
+        const { keysToRemove } = action;
+
+        removeProperties(copyState, keysToRemove);
+        break;
+
+      case 'clear':
+        copyState = {};
```

Comment:
Since stateCopy is now a constant, we need to come up with a new way to clear the object.

Reply:
There is a condition in the task description that if the type is clear you should create an empty state object. So I did it this way. In the previous task, I used a loop to delete all the properties.

Reply:
In the way you are doing it, it creates new reference for `copyState` object, and the old value is still somewhere there in the memory
So here is better to loop and delete properties, however it can be slow with large objects, but usually we don`t have those

Reply:
I understand that it creates a new reference for copyState. And I agree with you. But there is a condition in the description of this task, and we must adhere to it in our solution of the task.